### DISCUSSION TITLE
Components not mapped correctly under their subfolders

### DISCUSSION THREAD

**Discussion by telisjok** [nuxt/nuxt#17660] [Questions]
After updating Nuxt3 to latest version, I noticed that there is a change regarding the components inside subfolders in the `components` folder. 

Previously, by defining the components `dirs` in `nuxt.config`, the components where mapped correctly in the `components.d.ts` file under their correct subfolders.
Now, after updating to Nuxt latest version, components under subfolders have awkward names eg. `/components/forms/sign-in` will have the name `FormsSignIn` instead of `SignIn`.

I see that NUXT3 official DOCS don't provide any further info on how we can have the old behavior back.
Is it possible to keep the components under their subfolders and have the correct names? Currently my solution was to move every component out from their subfolders to the root `component` folder but this won't work for too long as soon as the project starts having more and more of them.

For example, there is a `SignIn` component for the sign in page layout and there is another `SignIn` component for the sign-in form. I don't want them to have names like `LayoutSignIn` and `FormSignIn` because everything is being ported from Storyblok and the names in the components there would look very awkward for the content editors.

Any suggestions would be highly appreciated.

**Comment by BlackyWolf** [comment]
Seconded.

I don't necessarily disagree with the naming convention in some cases, but it's made naming annoying, especially for teams who already have a pre-set naming policy either for that team specifically or mandated company wide. So being able to turn it off would be nice.

For anyone else wondering, a temporary work around is to import the components explicitly where you need them.
